add a script for finding popular, but unused projects

adamdecaf opened the pull request:
- `popular`: the more godoc importers the more popular 
- `unused`: long time since last commit? few issue updates/comments

Issue: https://github.com/gofrs/help-requests/issues/14

adamdecaf committed:
adding script to find popular, but unused projects

Issue: https://github.com/gofrs/help-requests/issues/14

acln0 commented on main.go:
```diff
@@ -0,0 +1,117 @@
+)
+
+var (
+	ctx = context.Background()
```
I think we should avoid this. For now, let's pass `context.Background()` to all things that need it, explicitly.

If the context we use changes in the future, it'll be easier to spot at a glance and adjust.

acln0 reviewed (changes requested):
Looks nice. I'd like to request a few minor adjustments before we merge, though.

Besides the comments in the review, please squash the PR to one commit too.

Thanks.

acln0 commented:
Oh, something else too. I only just now noticed that this vendors dependencies, but it is using modules as well. Isn't that strange? I thought modules and vendoring were incompatible.

adamdecaf commented:
@acln0 Thanks for the review. I dropped `vendor/`, but it's still used by `go build`. This script requires the internet already.

adamdecaf commented:
There are a lot of -1's returned still, so I should probably look into
that. I hope we can just make that refresh call and retry.

On Wed, Sep 12, 2018 at 2:19 PM, Adam Shannon <notifications@github.com>
wrote:

> @acln0 <https://github.com/acln0> Thanks for the review. I dropped vendor/,
> but it's still used by go build. This script requires the internet
> already.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/gofrs/help-requests/pull/29#issuecomment-420801529>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAHYdzxppj14aB6ISMfaaXpUHVfqbnUiks5uaXpMgaJpZM4Wl6eE>
> .
>

acln0 commented:
Thanks. This looks pretty good now.

I guess the next question is if we want to have a `main.go` in the repository root, or if we should move it to `cmd/something`.

Ping @theckman for review and further thoughts.

theckman commented on main.go:
```diff
@@ -0,0 +1,115 @@
+package main
```
Let's make this `cmd/finder/main.go` or something similar.

acln0 replied:
Maybe add a `// import "github.com/gofrs/help-requests/cmd/finder"` comment here too?

Perhaps a package-level comment would be nice too, like here https://github.com/golang/tools/blob/master/cmd/stringer/stringer.go

theckman reviewed (changes requested):
If the option is between using modules and vendoring, I'd prefer we do what we need to have a functioning vendor directory. I'd like to make sure the binary is buildable no matter the state of any of its dependencies.

Can you please update this PR to add a vendor directory that ensures the script will build?

adamdecaf commented:
Added `vendor/` back. Updated the script a bit. There are still a lot of `-1` godoc counts coming back. I think we should try a manual refresh on godoc.org to grab their count. Thoughts?
